File: Labs/lab01/types2.py
def count_type(arr):
    
    count = {}
   
    for value in arr:
        if type(value) is bool:
            t = "bool"
        elif type(value) is int:
            t = "int"
        elif type(value) is str:
            t = "str"
        else:
            t = "other"

        if t not in count:
            count[t] = 0  
        count[t] += 1
    return count
    

array = [1,"hi", True, 5, {}, "a", False]

counter = count_type(array)

print(counter)

File: Labs/lab01/types2.js
function count_type(array) {
    const count = {}; 

    for (const value of array) {
        let t;

        if (typeof value === "boolean") {
            t = "bool";
        } else if (typeof value === "number") {
            t = "number";
        } else if (typeof value === "string") {
            t = "string";
        } else {
            t = "other";
        }

        if (!(t in count)) {
            count[t] = 0;
        }

        count[t] += 1;
    }

    return count;
}

const array = [1, "hi", true, 5, {}, "a", false];

const counter = count_type(array);

console.log(counter);
